src/components/previewOverlay/PreviewOverlay.tsx

```diff
@@ -0,0 +1,131 @@
+import styled from "styled-components";
+import ChevronLeftSharpIcon from "@mui/icons-material/ChevronLeftSharp";
+import ChevronRightSharpIcon from "@mui/icons-material/ChevronRightSharp";
+import { Button, IconButton } from "@material-ui/core";
+import { callToWebGL } from "../../util/unityContextActions";
+import { useState } from "react";
+
+const PreviewRoot = styled.div`
+  display: absolute;
+  width: inherit;
+  height: 900px;
+  opacity: 1;
+  pointer-events: auto;
+`;
+const PreviewGrid = styled.div`
+  display: grid;
+  width: inherit;
+  height: inherit;
+  grid-template-columns: repeat(19, 1fr);
+  grid-template-rows: repeat(15, 1fr);
+`;
+
+const PreviewGridLeft = styled.div`
+  grid-column: 5 / span 1;
+  grid-row: 14 / span 1;
+  z-index: 2;
+  pointer-events: auto;
+  margin: 5%;
+  min-height: 2em;
+  min-width: 2em;
+`;
+
+const PreviewGridCenter = styled.div`
+  grid-column: 7 / span 7;
+  grid-row: 13 / span 1;
+  z-index: 2;
+  pointer-events: auto;
+  margin: 5%;
+  min-height: 1em;
+  min-width: inherit;
+  padding-left: 10%;
+  padding-right: 10%;
+`;
+
+const PreviewGridRight = styled.div`
+  grid-column: 15 / span 1;
+  grid-row: 14 / span 1;
+  z-index: 2;
+  pointer-events: auto;
+  margin: 5%;
+  min-height: 2em;
+  min-width: 2em;
+`;
+
+const ActionTabList = styled.div`
+  display: flex;
+  overflow-x: scroll;
+  align-items: center;
+`;
+const ActionTabListItem = styled.div`
+  min-height: 1em;
+  outline-style: solid;
+  outline-width: 0.1em;
+  outline-color: black;
+  margin: 1em;
+  align-self: center;
+`;
+
+const ActionTabSelectedListItem = styled.div`
+background-color: white;
+`;
+
+function PreviewOverlay({actionList}): JSX.Element {
+
+  const [selected, setCount] = useState(0);
+
+
+
+
+  return (
+    <PreviewRoot>
+      
+      {actionList[0] !== undefined ? 
+      <PreviewGrid>
+        <PreviewGridLeft>
+          <IconButton
+            style={{
+              minWidth: "100%",
+              minHeight: "100%",
+              outlineStyle: "solid",
+              outlineWidth: "0.1em",
+              outlineColor: "black",
+            }}
+            onClick={selected > 0 ? ()=>{setCount(selected - 1); callToWebGL(actionList[selected-1][0], actionList[selected-1][1]) } :undefined}

```

Comment:
nested ternary operation is a  code smell and causes confusion. It's better to simplify this into an independent statement.

Reply:
There are a few similar code smells in Overlay.tsx as well which i did not write. Should i correct those as well?

Reply:
all code smells are resolved